index-handlers/src/main/java/no/sikt/nva/nvi/index/UpdateIndexHandler.java

```diff
@@ -0,0 +1,135 @@
+package no.sikt.nva.nvi.index;
+
+import static no.sikt.nva.nvi.common.DatabaseConstants.SORT_KEY;
+import static no.sikt.nva.nvi.common.service.NviService.defaultNviService;
+import static no.sikt.nva.nvi.index.aws.OpenSearchClient.defaultOpenSearchClient;
+import static no.sikt.nva.nvi.index.utils.NviCandidateIndexDocumentGenerator.generateDocument;
+import static nva.commons.core.attempt.Try.attempt;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.DynamodbEvent;
+import com.amazonaws.services.lambda.runtime.events.DynamodbEvent.DynamodbStreamRecord;
+import com.amazonaws.services.lambda.runtime.events.models.dynamodb.OperationType;
+import java.net.URI;
+import java.util.Optional;
+import java.util.UUID;
+import no.sikt.nva.nvi.common.StorageReader;
+import no.sikt.nva.nvi.common.model.CandidateWithIdentifier;
+import no.sikt.nva.nvi.common.model.business.Candidate;
+import no.sikt.nva.nvi.common.service.NviService;
+import no.sikt.nva.nvi.index.aws.S3StorageReader;
+import no.sikt.nva.nvi.index.aws.SearchClient;
+import no.sikt.nva.nvi.index.model.NviCandidateIndexDocument;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.paths.UriWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UpdateIndexHandler implements RequestHandler<DynamodbEvent, Void> {
+
+    public static final String DOCUMENT_ADDED_MESSAGE = "Document has been added/updated";
+    public static final String DOCUMENT_REMOVED_MESSAGE = "Document has been removed";
+    private static final String CANDIDATE_TYPE = "CANDIDATE";
+    private static final String PRIMARY_KEY_DELIMITER = "#";
```

Comment:
Nitpicking: I think in publication-api we use ":" as delimiter in the resource database.